topostats/grainstats.py

```diff
@@ -0,0 +1,653 @@
+            # Find the cartesian extremities
+            x_min = np.min(rotated_points[:, 0])
+            x_max = np.max(rotated_points[:, 0])
+            y_min = np.min(rotated_points[:, 1])
+            y_max = np.max(rotated_points[:, 1])
+
+            if debug:
+                # Create plot
+                # FIXME : Make this a private method
+                fig = plt.figure(figsize=(8, 8))
+                ax = fig.add_subplot(111)
+
+                # Draw the points and the current simplex that is being tested
+                plt.scatter(x=remapped_points[:, 0], y=remapped_points[:, 1])
+                plt.plot(remapped_points[simplex, 0], remapped_points[simplex, 1], "#444444", linewidth=4)
+                plt.scatter(x=rotated_points[:, 0], y=rotated_points[:, 1])
+                plt.plot(rotated_points[simplex, 0], rotated_points[simplex, 1], "k-", linewidth=5)
+                print(rotated_points[simplex, 0], rotated_points[simplex, 1])
+
+                # Draw the convex hulls
+                for simplex in hull_simplices:
+                    plt.plot(remapped_points[simplex, 0], remapped_points[simplex, 1], "#888888")
+                    plt.plot(rotated_points[simplex, 0], rotated_points[simplex, 1], "#555555")
+
+                # Draw bounding box
+                plt.plot([x_min, x_min, x_max, x_max, x_min], [y_min, y_max, y_max, y_min, y_min], "#994400")
+                plt.savefig(path / ("bounding_rectangle_construction_simplex_" + str(simplex_index) + ".png"))
+
+            # Calculate the area of the proposed bounding rectangle
+            bounding_area = (x_max - x_min) * (y_max - y_min)
+
+            # If current bounding rectangle is the smallest so far
+            if smallest_bounding_area == None or bounding_area < smallest_bounding_area:
+                smallest_bounding_area = bounding_area
+                smallest_bounding_rectangle = (x_min, x_max, y_min, y_max)
+                aspect_ratio = (x_max - x_min) / (y_max - y_min)
+                smallest_bounding_width = min((x_max - x_min), (y_max - y_min))
+                smallest_bounding_length = max((x_max - x_min), (y_max - y_min))
+                # Enforce >= 1 aspect ratio
+                if aspect_ratio < 1.0:
+                    aspect_ratio = 1 / aspect_ratio
+
+        # Unrotate the bounding box vertices
+        RINVERSE = R.T
+        translated_rotated_bounding_rectangle_vertices = np.array(
+            ([x_min, y_min], [x_max, y_min], [x_max, y_max], [x_min, y_max])
+        )
+        translated_bounding_rectangle_vertices = []
+        for index, point in enumerate(translated_rotated_bounding_rectangle_vertices):
+            newpoint = RINVERSE @ point
+            # FIXME : As above can likely use np.append(, axis=0) here
+            translated_bounding_rectangle_vertices.append(newpoint)
+        translated_bounding_rectangle_vertices = np.array(translated_bounding_rectangle_vertices)
+
+        if debug:
```

Comment:
Best avoided in future - a debugger should be used instead.

Reply:
That was my hacky solution to not moving the plotting out to `topostats.plottingfuncs` as @SylviaWhittle said that its mainly code she was using whilst working through writing this class. Will investigate methods of having plots created on entering debugging and agree that all plotting functions should be in their own sub-module.